censustree/gravitontree/trie.go

```diff
@@ -182,18 +192,23 @@ func (t *Tree) treeWithRoot(root []byte) statedb.StateTree {
 	return t.store.TreeWithRoot(root)
 }
 
-// Dump returns the whole merkle tree serialized in a format that can be used on Import
-func (t *Tree) Dump(root []byte) (claims [][]byte, err error) {
+// Dump returns the whole merkle tree serialized in a format that can be used on Import.
+// Byte seralization is performed using bare message protocol, it is a 40% size win over JSON
+func (t *Tree) Dump(root []byte) ([]byte, error) {
 	t.updateAccessTime()
 	tree := t.treeWithRoot(root)
 	if tree == nil {
 		return nil, fmt.Errorf("dump: root not found %x", root)
 	}
+	dump := exportData{}
 	tree.Iterate(nil, func(k, v []byte) bool {
-		claims = append(claims, k)
+		ee := exportElement{Key: make([]byte, len(k)), Value: make([]byte, len(v))}
+		copy(ee.Key, k[:])
+		copy(ee.Value, v[:])
```

Comment:
document this copy? something like `it's not safe to hold on to the []byte values from Iterate`.